airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/demo/file_based/snowflake/SnowflakeStreamDestination.java

```diff
@@ -0,0 +1,70 @@
+package io.airbyte.integrations.destination.snowflake.demo.file_based.snowflake;
+
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig;
+import io.airbyte.integrations.base.destination.typing_deduping.TypeAndDedupeOperationValve;
+import io.airbyte.integrations.base.destination.typing_deduping.TyperDeduper;
+import io.airbyte.integrations.destination.snowflake.demo.file_based.iface.StreamDestination;
+import java.io.File;
+
+public class SnowflakeStreamDestination implements StreamDestination {
+
+  private final String originalStreamName;
```

Comment:
this is a big win over what exists today. Right now we have a ton of structs like [StreamId](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/StreamId.java) that bundle these together, because we need to be able to pass them everywhere. A stream-first approach lets us just put them into fields.